test/services/AuthenticatorServiceSpec.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import connectors.mocks.MockAuthenticatorConnector
+import helpers.TestConstants._
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.soletraderidentificationfrontend.models.SoleTraderDetailsMatching.{Deceased, Matched, Mismatch, NotFound}
+import uk.gov.hmrc.soletraderidentificationfrontend.services.AuthenticatorService
+import play.api.test.Helpers.{await, defaultAwaitTimeout}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class AuthenticatorServiceSpec extends AnyWordSpec with Matchers with MockAuthenticatorConnector {
+
+  object TestService extends AuthenticatorService(mockAuthenticatorConnector)
+
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  "matchSoleTraderDetails" should {
+    "return Right(Matched)" when {
+      "the provided details match those from authenticator" when {
+        "the enableSautrCheck is true and the sautr is provided" in {
+          mockMatchSoleTraderDetails(testSoleTraderDetails)(Future.successful(Right(testSoleTraderDetails)))
+
+          val result = await(TestService.matchSoleTraderDetails(testSoleTraderDetails, testJourneyConfig(enableSautrCheck = true)))
+
+          result mustBe Right(Matched)
+        }
+
+        "the enableSautrCheck is false and the sautr is not provided" in {
+          mockMatchSoleTraderDetails(testSoleTraderDetailsNoSautr)(Future.successful(Right(testSoleTraderDetailsNoSautr)))
+
+          val result = await(TestService.matchSoleTraderDetails(testSoleTraderDetailsNoSautr, testJourneyConfig()))
+
+          result mustBe Right(Matched)
+        }
+      }
+    }
```

Comment:
Could we add a scenario where the SAUTR does not match, but the enableSautrCheck boolean is false, so no check is performed? I know it's not a valid use case, but at a service level it is valid.